enclave-bridge/src/tests/test_cases.rs

```diff
@@ -0,0 +1,439 @@
+		let _enclave = register_sovereign_test_enclave(&enclave_signer);
+
+		// There are no events emitted at the genesis block.
+		System::set_block_number(1);
+		System::reset_events();
+
+		let hash = H256::from([1u8; 32]);
+		let extra_topics = vec![H256::from([2u8; 32]), H256::from([3u8; 32])];
+		let data = b"hello world".to_vec();
+
+		// publish with extra topics and data
+		assert_ok!(EnclaveBridge::publish_hash(
+			RuntimeOrigin::signed(enclave_signer.clone()),
+			hash,
+			extra_topics.clone(),
+			data.clone()
+		));
+
+		// publish without extra topics and data
+		assert_ok!(EnclaveBridge::publish_hash(
+			RuntimeOrigin::signed(enclave_signer.clone()),
+			hash,
+			vec![],
+			vec![]
+		));
+
+		let fingerprint = Teerex::sovereign_enclaves(&enclave_signer).unwrap().fingerprint();
+		let mut topics = extra_topics;
+		topics.push(fingerprint.into());
+
+		// Check that topics are reflected in the event record.
+		assert_eq!(
+			System::events(),
+			vec![
+				EventRecord {
+					phase: Phase::Initialization,
+					event: EnclaveBridgeEvent::PublishedHash {
+						fingerprint: fingerprint.into(),
+						hash,
+						data
+					}
+					.into(),
+					topics,
+				},
+				EventRecord {
+					phase: Phase::Initialization,
+					event: EnclaveBridgeEvent::PublishedHash {
+						fingerprint: fingerprint.into(),
+						hash,
+						data: vec![]
+					}
+					.into(),
+					topics: vec![fingerprint.into()],
+				},
+			]
+		);
+	})
+}
+
+#[test]
+fn publish_hash_with_unregistered_enclave_fails() {
+	new_test_ext().execute_with(|| {
+		let enclave_signer = AccountKeyring::Eve.to_account_id();
+
+		assert_err!(
+			EnclaveBridge::publish_hash(
+				RuntimeOrigin::signed(enclave_signer),
+				[1u8; 32].into(),
+				vec![],
+				vec![]
+			),
+			pallet_teerex::Error::<Test>::EnclaveIsNotRegistered
+		);
+	})
+}
+
+#[test]
+fn publish_hash_with_too_many_topics_fails() {
+	new_test_ext().execute_with(|| {
+		let enclave_signer = AccountKeyring::Eve.to_account_id();
+		let _enclave = register_sovereign_test_enclave(&enclave_signer);
+
+		let hash = H256::from([1u8; 32]);
+		let extra_topics = vec![
+			H256::from([0u8; 32]),
+			H256::from([1u8; 32]),
+			H256::from([2u8; 32]),
+			H256::from([3u8; 32]),
+			H256::from([4u8; 32]),
+			H256::from([5u8; 32]),
+		];
+
+		assert_err!(
+			EnclaveBridge::publish_hash(
+				RuntimeOrigin::signed(enclave_signer),
+				hash,
+				extra_topics,
+				vec![]
+			),
+			Error::<Test>::TooManyTopics
+		);
+	})
+}
+
+#[test]
+fn publish_hash_with_too_much_data_fails() {
+	new_test_ext().execute_with(|| {
+		let enclave_signer = AccountKeyring::Eve.to_account_id();
+		let _enclave = register_sovereign_test_enclave(&enclave_signer);
+
+		let hash = H256::from([1u8; 32]);
+		let data = vec![0u8; DATA_LENGTH_LIMIT + 1];
+
+		assert_err!(
+			EnclaveBridge::publish_hash(RuntimeOrigin::signed(enclave_signer), hash, vec![], data),
+			Error::<Test>::DataTooLong
+		);
+	})
+}
+
+#[test]
+fn update_ipfs_hash_works() {
```

Comment:
I don't understand `ipfs_hash` in this context.

Reply:
me neither. just ported old tests

Reply:
The test is identical to `confirm_processed_parentchain_block_works`, so we should remove this one. and the other one can be renamed to `confirm_processed_parentchain_block_fails_for_not_registered_enclave`.

Reply:
I checked. the test names were wrong and one was redundant. fixed